### PR TITLE
enhance: ensure that each element stack in microlearning can only be submitted once

### PR DESCRIPTION
**Pull request by sjschlapbach**

### CODE REVIEW

File: packages/graphql/src/schema/query.ts [hunk 1]
```diff
@@ -279,6 +279,17 @@ export const Query = builder.queryType({
         },
       }),
 
+      getPreviousStackEvaluation: t.field({
```
Comment (rschlaefli) [inline packages/graphql/src/schema/query.ts:282]:
```suggestion
      getPreviousStackEvaluation: asParticipant.field({
```

File: packages/graphql/src/services/practiceQuizzes.ts [hunk 2]
```diff
@@ -2160,6 +2164,220 @@ function combineStackStatus({
   return prevStatus
 }
 
+export async function getPreviousStackEvaluation(
+  { stackId }: { stackId: number },
+  ctx: Context
```
Comment (rschlaefli) [inline packages/graphql/src/services/practiceQuizzes.ts:2169]:
```suggestion
  ctx: ContextWithUser
```

?

File: packages/graphql/src/services/practiceQuizzes.ts [hunk 3]
```diff
@@ -2179,6 +2397,32 @@ export async function respondToElementStack(
   { stackId, courseId, responses, stackAnswerTime }: RespondToElementStackInput,
   ctx: Context
 ) {
+  // if the element stack is part of a microlearning and the student has already responses to it, ignore this submission
+  if (ctx.user?.sub) {
+    const stack = await ctx.prisma.elementStack.findUnique({
+      where: { id: stackId },
```
Comment (rschlaefli) [inline packages/graphql/src/services/practiceQuizzes.ts:2403]: element stack type

**Review by rschlaefli — APPROVED** [review]

File: packages/graphql/src/services/practiceQuizzes.ts (hunk 3 above)
Comment (sjschlapbach) [reply packages/graphql/src/services/practiceQuizzes.ts:2403]: done

**Pull request merged**